### DISCUSSION TITLE
SERVER_PUBLIC not working on 0.147

### DISCUSSION THREAD

**Discussion by Silvers12** [lloesche/valheim-server-docker#139] [Q&A] [answered]
Hi,
I use SERVER_PUBLIC="false" for my server. But since the 0.147 if i use false my server don't show in the Steam server list...
I need to pass true for join it.

Someone have the same issue ?

Thanks for the work ;)

**Comment by InB4DevOps** [comment]
Well, that's what the flag is for. When you set it to false then it won't be visible in the server list.

**Reply by Silvers12** [reply]
Yes in the game list, but not in steam server.
"If you change the public to false, people will have to join using the Steam server browser, found under View > Servers on the Steam client in the Favorites tab, then using the "Add A Server" button in the bottom right."
And before the 0.147 it work like that. Not list in game but visible in steam server.

**Comment by InB4DevOps** [comment]
Did you add it to your steam server list manually?

Also it seems you now also need to forward port 2455 & 2456 to your server. 

https://steamcommunity.com/app/892970/eventcomments/3105766250724825196?snr=1_2108_9__2107&ctp=16#c3105766250725225492

**Reply by Silvers12** [reply]
Yes i add it manually and it work until the last update. I didn't open the 2455 just 2456 - 2458 udp like the docker cmd say "-p 2456-2458:2456-2458/udp". 
I can try to add the 2455

**Reply by InB4DevOps** [reply]
adding 2455 did the trick for me. I had the same connection problem prior to adding 2455.

**Reply by Silvers12** [reply]
I have open the 2455 and add it into the cmd docker -p 2455-2458:2455-2458/udp but always the same issue.

docker run -d --name valheim-server -p 2455-2458:2455-2458/udp -v /home/steam/valheim-server/config:/config -v /home/steam/valheim-server/data:/opt/valheim -e SERVER_NAME="MyName" -e WORLD_NAME="MyWorld" -e SERVER_PASS="********" -e SERVER_PUBLIC="false" -e TZ="Europe/Paris" lloesche/valheim-server

**Comment by lloesche** [comment, marked as answer]
Looks like this is a wider spread issue. Also there is no need to open 2455 ports have stayed the same as before. It just so happens that for some folks after restarting the container it starts to work again.
Someone in another issue linked this article https://www.pcgamesn.com/valheim/dedicated-server-disconnected

@Silvers12 also they seem to have changed the way private/public servers work. The changelog says
```
Added -public 1/0 flag to dedicated server again, Allows players to host local lan only servers
```
So public 0 seems to be for LAN only games now.

**Reply by Silvers12** [reply]
Ok good to know. 
Thx for the answer.

And thx for the work ;)

**Comment by InB4DevOps** [comment]
Didn't mean to cause confusion here. I read on Steam in comments that forwarding 2455 helped so I gave it a shot and it worked for me after that. Must have been a coincidence then.

**Reply by Silvers12** [reply]
Don't worry, you answered me and tried to find me a solution and I say thank you for that.
